ansible_collections/arista/avd/docs/porting-guides/5.x.x.md

```diff
@@ -514,6 +514,107 @@ In AVD 4.0.0, we had "switched" as the default value for `ethernet_interfaces[].
+
+With AVD 5.0.0, the dependency on the `type` key for rendering the configuration of `vlan_id` in ethernet and port-channel interfaces has been removed.
+
+```diff
+-type: l2dot1q
+vlan_id: 10
+```
+
+#### `switchport` configuration and documentation
+
+In AVD 4.0.0, the configuration and documentation for `switchport` in ethernet and port-channel interfaces were renderend only if `ethernet_interfaces[].type` or `port_channel_interfaces[].type` was set to `switched` along with the keys `mode`, `vlans`, `native_vlan`, `native_vlan_tag`, `trunk_groups`, `phone`, `vlan_translations`, `trunk_private_vlan_secondary`, `pvlan_mapping`.
+The `no switchport` command was renderend if `ethernet_interfaces[].type` or `port_channel_interfaces[].type` was set to `routed`.
+
+With AVD 5.0.0, a new key for `switchport` has been introduced under ethernet and port-channel interfaces data-model, which generates the configuration and documentation for `switchport` independent of the `type` key.
+The keys `mode`, `vlans`, `native_vlan`, `native_vlan_tag`, `trunk_groups`, `phone`, `vlan_translations`, `trunk_private_vlan_secondary`, `pvlan_mapping` are deprecated, to be removed in AVD 6.0.0.
+
+```diff
+-type: switched
+-mode: trunk
+-vlans: '1-5'
+-native_vlan: 20
+-native_vlan_tag: true
+-trunk_groups: ['group1', 'group2']
+-trunk_private_vlan_secondary: true
+-pvlan_mapping: 20-30
+-  phone:
+-    vlan: 35
+-    trunk: tagged
+-vlan_translations:
+-  - from: 10
+-    to: 20
+-    direction: in
+-  - from: 23
+-    to: 30
+-    direction: out
+-  - from: 1
+-    to: 5
+-    direction: both
++switchport:
++  mode: trunk
++  trunk:
++    allowed_vlan: 1-5
++    native_vlan: 20
++    native_vlan_tag: true
++    trunk_groups: ['group1', 'group2']
++    private_vlan_secondary: true
++  pvlan_mapping: 20-30
++  phone:
++    vlan: 35
++    trunk: tagged
++  vlan_translations:
++    direction_in:
++      - from: 10
++        to: 20
++    direction_out:
++      - from: 23
++        to: 30
++    direction_both:
++      - from: 1
++        to: 5
+```
+
+#### `encapsulation` configuration and documentation
+
+In AVD 4.0.0, the configuration and documentation for `encapsulation` in ethernet and port-channel interfaces were renderend only if `ethernet_interfaces[].type` or `port_channel_interfaces[].type` was set to `l2dot1q/l3dot1q` along with the keys `encapsulation_dot1q_vlan` and `encapsulation_vlan`.
+
+With AVD 5.0.0, a new key for `encapsulation_dot1q` has been introduced and the schema for `encapsulation_vlan` has been modified under ethernet and port-channel interfaces data-model, which generates the configuration and documentation for `encapsulation` independent of the `type` key.
+The keys `encapsulation_dot1q_vlan`, `encapsulation_vlan.client.dot1q`, `encapsulation_vlan.client.unmatched`, `encapsulation_vlan.network.dot1q` and `encapsulation_vlan.network.client` are deprecated, to be removed in AVD 6.0.0.
+
+```diff
```

Comment:
Same with some ethernet_interfaces context. Remember this should be a subinterface so `name: Ethernet1.20`